### PR TITLE
Differentiate between a sequence's database schema and model schema

### LINKED ISSUE #28223: HiLo sequences are being dropped when specifying a default schema
I'm having an issue where with each new migration file being generated our HiLo sequences are being dropped. This only happens if I add a default schema to the DbContext. This is the result when creating new migrations:

image

image

Here is the necessary code to reproduce this issue:

```C#
protected override void OnModelCreating(ModelBuilder modelBuilder)
 {
        modelBuilder.HasDefaultSchema("MyDefaultSchema");
        
         modelBuilder.HasSequence<int>("HiLoTestSeq", "MyDefaultSchema")
            .StartsAt(10) // Make provision for defaults
            .IncrementsBy(5);
}
   
public class TestSchemaEntity
{
    public int Id { get; private set; }
}

internal class TestSchemaEntityTypeConfiguration : IEntityTypeConfiguration<TestSchemaEntity>
{
    public void Configure(EntityTypeBuilder<TestSchemaEntity> builder)
    {
        builder.ToTable("TestSchemaTable", "MyDefaultSchema");

        builder.HasKey(r => r.Id);

        builder.Property(r => r.Id).UseHiLo("HiLoTestSeq", "MyDefaultSchema");
    }
}
```

```

EF Core version:
Database provider:  Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET 6.0
Operating system: Window 11
IDE:  Visual Studio 2022 16.3

### PR DESCRIPTION
**Pull request by bricelam**
Fixes #28223

### CODE REVIEW

File: src/EFCore.Relational/Metadata/RuntimeSequence.cs [hunk 1]
```diff
@@ -65,6 +69,12 @@ public class RuntimeSequence : AnnotatableBase, ISequence
     /// </summary>
     public virtual string Name { get; }
 
+    /// <summary>
+    ///     Gets the metadata schema of the sequence.
+    /// </summary>
+    public virtual string? ModelSchema
+        => _modelSchemaIsNull ? null : _schema;
```
Comment (bricelam) [inline src/EFCore.Relational/Metadata/RuntimeSequence.cs:76]: We could probably just throw here too...

File: test/EFCore.Design.Tests/Migrations/ModelSnapshotSqlServerTest.cs [hunk 2]
```diff
@@ -1278,6 +1278,42 @@ public virtual void Sequence_is_stored_in_snapshot_as_fluent_api()
...
+                .HasDefaultSchema("dbo")
+                .Entity("Entity").Property<int>("Id").UseHiLo(schema: "dbo"),
+            AddBoilerPlate(@"
+            modelBuilder
+                .HasDefaultSchema(""dbo"")
+                .HasAnnotation(""Relational:MaxIdentifierLength"", 128);
+
+            SqlServerModelBuilderExtensions.UseIdentityColumns(modelBuilder);
+
+            modelBuilder.HasSequence(""EntityFrameworkHiLoSequence"", ""dbo"")
```
Comment (bricelam) [inline test/EFCore.Design.Tests/Migrations/ModelSnapshotSqlServerTest.cs:1294]: Before this change, we wouldn't scaffold the schema here causing two sequences to be added to the model (the other by UseHiLo)

**Review by AndriySvyryd — APPROVED** [review]

**Comment by dogahasdemir** [comment]
Hi,

Is there any update about this matter? After switching from EF Core 5.0 to EF Core 7.0, our migrations started to try drop the non existing Sequence "EntityFrameworkHiLoSequence"  which is the default Sequence coming from EF Core library itself.
We do not specify anydefault schema for the model. 

Thanks

**Pull request merged**